### PR TITLE
Adds okta-provider backend module for the auth backend

### PR DESCRIPTION
**Pull request by Josh-Uvi**
## Hey, I just made a Pull Request!

This relates to #19476

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by ataylorme** [comment]
@Josh-Uvi I was working on making `scope` customizable in #20580 but it might make sense to add that along with the Auth Backend Module Migration. What do you think? If you agree you can add it based on that PR or I can fork from your repo/branch and make a new PR against yours.

**Comment by Josh-Uvi** [comment]
Hi @ataylorme, I agree. You can fork my repo/branch and create a PR against it.
> @Josh-Uvi I was working on making `scope` customizable in #20580 but it might make sense to add that along with the Auth Backend Module Migration. What do you think? If you agree you can add it based on that PR or I can fork from your repo/branch and make a new PR against yours.

**Comment by ataylorme** [comment]
Thanks for being open to it @Josh-Uvi, I made a PR https://github.com/Josh-Uvi/backstage/pull/1 for your review.

**Comment by ataylorme** [comment]
Please do not close this PR, it is waiting on review

**Comment by Josh-Uvi** [comment]
> Thanks for being open to it @Josh-Uvi, I made a PR [Josh-Uvi#1](https://github.com/Josh-Uvi/backstage/pull/1) for your review.

Hi @ataylorme, sorry for the slow response, I've been hands-off all last week. Your PR has been merged with my branch.

File: plugins/auth-backend-module-okta-provider/README.md
Comment (camilaibs) [inline plugins/auth-backend-module-okta-provider/README.md:3]: Suggestion: I think adopters would appreciate more documentation about the configuration options, as well as an example of how to install this module.
What are your thoughts on this?

File: plugins/auth-backend-module-okta-provider/src/resolvers.ts
Comment (camilaibs) [inline plugins/auth-backend-module-okta-provider/src/resolvers.ts:33]: Question: There was a `emailMatchingUserEntityAnnotation` resolver on the old provider file... Just confirming, does that no longer make sense 🙂?

File: plugins/auth-backend-module-okta-provider/package.json
Comment (camilaibs) [inline plugins/auth-backend-module-okta-provider/package.json:4]: Quick question: I was expecting this version to be 0.0.0 since it is a new package, was this version generated automatically?

File: plugins/auth-backend/package.json
Comment (camilaibs) [inline plugins/auth-backend/package.json:46]: It seems that the "_@davidzemon/passport-okta-oauth_" is no longer used. Could you check if there is any other dependency cleanup we can do?

**Review by camilaibs — CHANGES_REQUESTED** [review]
Thanks for opening this pull request 🙌🏻, this looks nice!
I left minor comments below, I'd also appreciate a second look from @backstage/maintainers 🙂 .

File: plugins/auth-backend-module-okta-provider/CHANGELOG.md
Comment (freben) [reply plugins/auth-backend-module-okta-provider/CHANGELOG.md:1]: Correct. Changesets are generated by the release flows.

File: plugins/auth-backend-module-okta-provider/src/resolvers.ts
Comment (freben) [reply plugins/auth-backend-module-okta-provider/src/resolvers.ts:33]: Is it covered by the common email based one perhaps? https://github.com/backstage/backstage/blob/1b3fa23361f1af7c6054f15f1e5e5ebd1710ea9f/plugins/auth-node/src/sign-in/commonSignInResolvers.ts#L29

**Comment by freben** [comment]
Also remember to DCO sign every single commit! See the build details below or the bot comment above for details on how to fix it retroactively

File: plugins/auth-backend-module-okta-provider/config.d.ts
Comment (freben) [inline plugins/auth-backend-module-okta-provider/config.d.ts:21]: this can be removed from the config.d.ts file in auth-backend now

File: .changeset/blue-oranges-unite.md
Comment (freben) [inline .changeset/blue-oranges-unite.md:2]: You need to make a separate changeset for auth-backend too, saying that the okta implementation was moved to the new module

**Comment by freben** [comment]
Oh one more thing. You need to run `yarn build:api-reports` and commit the api report changes with the PR too.

File: plugins/auth-backend-module-okta-provider/config.d.ts
Comment (Josh-Uvi) [reply plugins/auth-backend-module-okta-provider/config.d.ts:21]: done

File: plugins/auth-backend-module-okta-provider/src/resolvers.ts
Comment (Josh-Uvi) [reply plugins/auth-backend-module-okta-provider/src/resolvers.ts:33]: @camilaibs good shout! I used the `auth-backend-module-gitlab-provider` as a guide, so not sure it still makes sense to keep it. @freben I think it's covered by the `commonSignInResolvers`. 
I will revert to using emailMatchingUserEntityAnnotation to keep it congruous with the old provider

File: plugins/auth-backend/package.json
Comment (Josh-Uvi) [reply plugins/auth-backend/package.json:46]: removed `@davidzemon/passport-okta-oauth` from auth-backend

File: plugins/auth-backend-module-okta-provider/README.md
Comment (Josh-Uvi) [reply plugins/auth-backend-module-okta-provider/README.md:3]: @camilaibs I agree, and I have included an example of how to use this module in the readME.

File: plugins/search-backend/src/service/router.test.ts
Comment (tikabom) [inline plugins/search-backend/src/service/router.test.ts:83]: left in here by mistake?

**Review by vinzscam — CHANGES_REQUESTED** [review]
Hi @Josh-Uvi, something went wrong probably when you resolved some conflicts against master. The branch contains a lot of commits which should be there

**Comment by Josh-Uvi** [comment]
> Hi @Josh-Uvi, something went wrong probably when you resolved some conflicts against master. The branch contains a lot of commits which should be there

Hi @vinzscam, yea I noticed. I will close this PR, and then create a new branch and PR off the master and move the commits there.

**Comment by Josh-Uvi** [comment]
@camilaibs @vinzscam I opened a new PR - #21325

**Pull request closed without merging**
